great_expectations/expectations/expectation.py

```
@@ -433,6 +480,24 @@ def _validate_result_format(cls, result_format: ResultFormat | dict) -> ResultFo
             )
         return result_format
 
+    @pydantic.root_validator(pre=False)
+    def _transform_legacy_row_condition(cls, values: dict) -> dict:
+        row_condition = values.get("row_condition")
+        condition_parser = values.get("condition_parser")
+        is_great_expectations_condition_parser = (
+            condition_parser is not None
+            and condition_parser
+            in [
+                CONDITION_PARSER_GREAT_EXPECTATIONS,
+                CONDITION_PARSER_GREAT_EXPECTATIONS_DEPRECATED,
+            ]
+        )
+        if isinstance(row_condition, str) and is_great_expectations_condition_parser:
+            condition_obj = _convert_string_to_condition(row_condition)
+            values["row_condition"] = condition_obj
+            values.pop("condition_parser", None)
```

If we pop the condition, do we have a mechanism to transform back to the old type if serving old versions of v1 core?

---

Could you clarify in what cases we’d actually need this backward compatibility?

---

Yeah, maybe my understanding was wrong here, but my thinking has been that there's v1 core versions before we supported the Condition class. So I think we want to do what your PR does, but then when returning expectations to those old versions of gx core, we'd have to convert back to something that version understands. I figured we'd want to convert back to the condition type they used, but maybe that's not needed?

---

It seems like this is not going to be something we would want to support, but maybe @NathanFarmer can confirm this.

---

@tyler-hoffman we discussed converting back, but in [the design doc](https://greatexpectations.atlassian.net/wiki/spaces/gx/pages/2027487236/Multiple+Row+Conditions+TDD?focusedCommentId=2053177383) and pointing session decided it would be better to raise an error and ask the user to upgrade Core (I have started add that change [here](https://github.com/greatexpectationslabs/mercury/pull/5393/files)).